gemfirexd/client/src/main/java/io/snappydata/jdbc/TomcatConnectionPool.java

```diff
@@ -23,149 +23,222 @@
+        String.valueOf(Connection.TRANSACTION_NONE));
+
+    final String key;
+    final String defValue;
+
+    PoolProps(String propKey, String defValue) {
+      this.key = propKey;
+      this.defValue = defValue;
     }
 
-    /**
-     * Initialize the Data Source with the Connection pool and returns the connection
-     *
-     * @return java.sql.Connection
-     * @throws SQLException
-     */
-    public Connection getConnection() throws SQLException {
-        return datasource.getConnection();
+    public static List<String> getKeys() {
+      PoolProps[] props = PoolProps.values();
+      List<String> keys = new ArrayList<>(props.length);
+      for (PoolProps prop : props) {
+        keys.add(prop.key);
+      }
+      return keys;
     }
-
-    /**
-     * Initializes the Object with passed on properties.
-     *
-     * @param props set of properties to be used for pool connections that includes
-     *              pool properties as well as connection properties
-     */
-    public TomcatConnectionPool(Properties props) {
-
-        List<String> listPoolPropKeys = PoolProps.getKeys();
-
-        PoolProperties poolProperties = setPoolProperties(props);
-
-        // Filtering out the pool properties and creating string of
-        // connection properties to pass on.
-        Set<String> keys = props.stringPropertyNames();
-        String connectionProperties = keys.stream().filter(x -> !listPoolPropKeys.contains(x))
-                .map(i -> i + "=" + props.getProperty(i))
-                .collect(Collectors.joining(";"));
-
-        poolProperties.setConnectionProperties(connectionProperties);
-        datasource = new DataSource();
-        datasource.setPoolProperties(poolProperties);
+  }
+
+  private DataSource datasource;
+
+  private static final ConcurrentMap<Properties, TomcatConnectionPool> poolMap =
+      new ConcurrentHashMap<Properties, TomcatConnectionPool>();
+
+  /**
+   * Initialize the Data Source with the Connection pool and returns the connection
+   *
+   * @return java.sql.Connection
+   * @throws SQLException
+   */
+  private Connection getConnection() throws SQLException {
+    return datasource.getConnection();
+  }
+
+  public static Connection getConnection(Properties properties) throws SQLException {
+    TomcatConnectionPool pool = poolMap.computeIfAbsent(properties,
+        props -> new TomcatConnectionPool(props));
+    return pool.getConnection();
+  }
+
+  /**
+   * Initializes the Object with passed on properties.
+   *
+   * @param prop
+   */
+  public TomcatConnectionPool(Properties prop) {
+
+    List<String> listPoolPropKeys = PoolProps.getKeys();
+
+    PoolProperties poolProperties = getPoolProperties(prop);
+
+    // Filtering out the pool properties and creating string of
+    // connection properties to pass on.
+    Set<String> keys = prop.stringPropertyNames();
+    String connectionProperties = keys.stream().filter(x -> listPoolPropKeys.contains(x))
```

Comment:
This should be "!listPoolPropKeys.contains(x)" that is all non-pool properties are connection properties and passed through to DriverManager.connect. I had fixed this on snappy/master.

Reply:
Corrected it, to allow only non-pool (which is connection) properties only allowed to pass in.